File: src/main/java/com/example/electronico/service/interfaces/CartService.java
package com.example.electronico.service.interfaces;

import com.example.electronico.exception.ProductNotFoundException;
import com.example.electronico.exception.UserNotFoundException;
import com.example.electronico.model.Cart;
import com.example.electronico.model.Order;
import com.example.electronico.model.User;

public interface CartService {
    Cart update(Cart cart) throws ProductNotFoundException;
    User pay(Order order, Long userId) throws UserNotFoundException;
}

File: src/main/resources/electronico/src/app/service/cart.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Cart} from "../model/cart.model";
import {Order} from "../model/order.model";

@Injectable({
  providedIn: 'root'
})
export class CartService {

  constructor(private http: HttpClient) { }

  update(cart: Cart): Observable<any> {
    return this.http.put('http://localhost:8084/api/cart/update', cart);
  }

  pay(order: Order, userId: number): Observable<any> {
    return this.http.post('http://localhost:8084/api/cart/pay/' + userId, order);
  }
}
